examples/architectureComponentsExample/src/main/java/io/realm/examples/arch/ExampleViewModel.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.arch;
+
+import android.arch.lifecycle.ViewModel;
+
+import io.realm.Realm;
+import io.realm.examples.arch.livemodel.LiveRealmObject;
+import io.realm.examples.arch.model.Person;
+
+
+public class ExampleViewModel extends ViewModel {
```

Comment:
Honestly I don't really understand the `Repository` layer. I think data access and loading remote data are separate in the sense that you might need to access the db to know if something *has to be* downloaded, but the whole "get it from network directly" thing is a mess!

Just observe the db and then write into the db from network call, if you do that then all data will update everywhere else.

Binding the two data sources together is a recipe for disaster, but maybe that's just what I think?